src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  const stateClone = { ...state };
+
+  for (const action of actions) {
+    const { extraData, type, keysToRemove } = action;
+
+    switch (type) {
+      case 'clear':
+        for (const key in stateClone) {
+          delete stateClone[key];
+        }
+        break;
+
+      case 'addProperties':
+        Object.assign(stateClone, extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          delete stateClone[key];
+        }
+        break;
+    }
+
```

Comment:
There is no default case in the switch statement, please check checklist.md.
`3. [CODE STYLE]: switch/case should always have default case for error handling.`